pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -420,6 +434,17 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 			return false, fmt.Errorf("selectedNode annotation value %q not set to scheduled node %q", selectedNode, pod.Spec.NodeName)
 		}
 
+		// If the PVC is bound to a PV, check its node affinity
+		if pvc.Spec.VolumeName != "" {
+			pv, err := b.pvCache.GetPV(pvc.Spec.VolumeName)
+			if err != nil {
+				return false, fmt.Errorf("failed to get pv %q from cache: %v", pvc.Spec.VolumeName, err)
+			}
+			if err := volumeutil.CheckNodeAffinity(pv, node.Labels); err != nil {
+				return false, fmt.Errorf("pv %q node affinity doesn't match node %q: %v", pv.Name, node.Name, err)
```

Comment:
I would say that the main point of having this check here is to prevent the scheduler from scheduling the pod rather than leaving the job to the kubelet.